scripts/globals/gobbiemysterybox.lua

```diff
@@ -18,6 +18,97 @@ local costs =
     [5] = 10,
     [6] = 50
 }
+local keyToDial =
+{
+    [8973] = 6,  -- special dial
+    [9217] = 9,  -- abjuration
+    [9218] = 10, -- fortune
+  --[????] = 11, -- furnishing
+    [9274] = 13, -- anniversary
+}
+local abjurationItems =
+{
+    1314,1315,1316,1317,1318, -- dryadic
+    1319,1320,1321,1322,1323, -- earthen
+    1324,1325,1326,1327,1328, -- aquarian
+    1329,1330,1331,1332,1333, -- martial
+    1334,1335,1336,1337,1338, -- wyrmal
+    1339,1340,1341,1342,1343, -- neptunal
+    1441,1442,                -- food
+    2429,2430,2431,2432,2433, -- phantasmal
+    2434,2435,2436,2437,2438, -- hadean
+    3559,3560,3561,3562,3563, -- corvine
+    3564,3565,3566,3567,3568, -- supernal
+    3569,3570,3571,3572,3573, -- transitory
+    3574,3575,3576,3577,3578, -- foreboding
+    3579,3580,3581,3582,3583, -- lenitive
+    8762,8763,8764,8765,8766, -- bushin
+    8767,8768,8769,8770,8771, -- vale
+    8772,8773,8774,8775,8776, -- grove
+    8777,8778,8779,8780,8781, -- triton
+    8782,8783,8784,8785,8786, -- shinryu
+    8787,8788,8789,8790,8791, -- abyssal
+    9105,9106,9107,9108,9109, -- cronian
+    9110,9111,9112,9113,9114, -- arean
+    9115,9116,9117,9118,9119, -- jovian
+    9120,9121,9122,9123,9124, -- venerian
+    9125,9126,9127,9128,9129, -- cyllenian
+}
+local fortuneItems =
+{
+    5737,5736, -- alexandrite
+    6180,6183,6532, -- pluton
+    6181,6184,6535, -- beitetsu
+    6182,6185,6534, -- rift boulder
+    5854,5855,5856,5857,5858, -- frayed pouches
+    5109,5110,5111,5112,5946,5947,6264,6345,6346,6370,6486,6487,6488 -- frayed sacks
+}
+local anniversaryItems =
+{
+    9274 -- just give them back their key until this table can be populated
+}
+local gobbieJunk =
+{
+    507,
+    508,
+    510,
+    511,
+    566,
+    568,
+    1239,
+    1868,
+    2542,
+    2543,
+    4539,
+    4543,
+    9777,
+    15203,
+    18180,
+    19220
+}
+tpz.mystery.onTrade = function (player, npc, trade, events)
+    if not trade then
+        return false
+    elseif trade:getItemCount() == 1 then
+        local keyID = trade:getItemId(0)
+        if keyID == nil or keyID < 1 or keyID > 65535 or trade:getItemCount() ~= 1 or trade:getSlotQty(0) ~= 1 then
+            return false
+        else
+            if player:getFreeSlotsCount() == 0 then
+                player:startEvent(events.FULL_INV, keyID, keyToDial[keyID])
+            elseif keyToDial[keyID] ~= 0 then
+                if player:getFreeSlotsCount() == 0 then
+                    player:startEvent(events.FULL_INV, keyID, keyToDial[keyID])
+                else
```

Comment:
Also don't need to check ` if player:getFreeSlotsCount() == 0 then` here on L100 because we already did so on L97 (so L100-L102 are impossible to reach).